Start to add operator tokens support

samverrall opened the pull request:
Added support for the '=', '>', '<', '|' operators.

samverrall committed:
Start to add operator tokens support

polyscone reviewed (changes requested):
It looks like you're missing a test for `<` (less) in `lexer_test.go`, so it's probably worth adding that quickly.

I've also spotted a bug that the tests don't catch. If you were to consume multiple tokens in a row you'd get stuck on the operators you've added because you never call `l.read()` or `l.readWhile()` to advance through the input.

To catch the bug in tests I'd suggest changing the `t.Run()` function in `lexer_test.go` to:
```go
t.Run(tc.name, func(t *testing.T) {
	l.Load(strings.NewReader(tc.input))

	if got := l.Consume(); got != tc.want {
		t.Errorf("want %v, got %v", tc.want, got)
	}

	if want, got := token.EOF, l.Consume().Kind; got != want {
		t.Errorf("want %v, got %v", want, got)
	}
})
```
So we'd just always check that the final token is an `EOF`.

You also don't need to hardcode the lexeme for operator tokens when you call `l.newToken()`. For example, you could just pass in `string(next)`.

samverrall committed:
Add missing testcase, clean up switch statement

samverrall commented:
Added in the test case for `<` in `lexer_test.go`.

Improved `lexer_test.go` with your suggestion in new commit.

Removed hardcoded lexeme and updaed to use `string(next)`.
